[WORK IN PROGRESS] review readme (#21)
MeteoSwiss/opendata-forecast-data: README.md

### 2.4 3D Grid Structure and Representation
The model data is structured on both a horizontal and vertical grid.

#### 2.4.1 Vertical Grid

The vertical grid above the surface is a height-based coordinate system that describes terrain-following model levels. The closer the levels are to
the surface, the narrower the layers they define, as shown in the image below. The model levels gradually change into levels of constant height as the distance from the surface increases.

> Each grid box is delimmited at the top and bottom by so-called half levels of the grid, while the full levels are aligned with the center of the grid box.

Reviewer: ```suggestion
Each grid box is delimited at the top and bottom by so-called half levels of the grid, while the full levels are aligned with the center of the grid box.
```

> The vertical grid uses a Lorenz-type staggering, meaning that some parameters are defined at full levels and others, e.g. the vertical velocity `W`, at full levels.

Reviewer: ```suggestion
The vertical grid uses a Lorenz-type staggering, meaning that some parameters are defined at full levels and others, e.g. the vertical velocity `W`, at half levels.
```

There are 81 discrete half levels and 80 full levels in our data. The levels are numbered from top to bottom.

<div align=center>
<img src="Images/VerticalLayers.png" width="550"/>

Illustration of ICON's vertical levels, Working with the ICON Model 2024, Figure 3.2
</div>

> In addition to the vertical grid above the surface (half and full levels), there is also a grid below the land surface to describe parameters as the soil temperature for example.

Reviewer: ```suggestion
In addition to the vertical grid above the surface (half and full levels), there is also a grid below the land surface to describe parameters such as the soil temperature for example.
```

3. Compare the two hash values. If they match, your forecast data file is safe to use.

Once the file is verified, you can proceed with decoding the GRIB file, using e.g. the instructions in [2.8.2 Decoding GRIB Files with ecCodes](#282-decoding-grib-files-with-eccodes).

### 2.7 Accessing Static Grid Information: Height, Longitude, and Latitude

Besides the current forecast files, each collection contains two static files. They store constant information about the height of the half levels (HHL) in the vertical grid and
the center point coordinates of each triangle on the horizontal grid.

> ❗ **NOTE**: The forecast GRIB files contain no information on height, longitude and latitude. They have to be determined via the static vertical and horizontal grid parameter files.

#### 2.7.1 Accessing Vertical Grid Parameters

> In the static vertical file, the heights of the half levels of the vertical grid are provided in meters above mean sea level. In order to associate a value from a data file (for a given parameter) to a height in meters above sea level, follow the steps below:

Reviewer: Thanks for catching the typo!!

```
curl -X GET https://data.geo.admin.ch/api/stac/v1/collections/ch.meteoschweiz.ogd-forecasting-icon-ch1/assets
```
2. Locate the `href` field under `assets` in `id: horizontal_constants_icon-ch1-eps.grib2` and copy the pre-signed URL.
3. Download the file with:
```
wget -O <desired_filename> “<pre-signed URL>”
```

> 4. Once the static GRIB file is downloaded, ensure that the `uuidOfHGrid` (Universally Unique Identifier for the horizonal grid) key in the data file matches the one in the static horizontal file.

Reviewer: ```suggestion
4. Once the static GRIB file is downloaded, ensure that the `uuidOfHGrid` (Universally Unique Identifier for the horizontal grid) key in the data file matches the one in the static horizontal file.
```

### 2.8 Reading Forecast Files Using ecCodes

Once you have the desired GRIB files, you need a tool to decode the GRIB records. Each GRIB record holds the data for one parameter
at one time and at one level. More information about the data format can be found in the Manual on Codes published by WMO.
We recommend installing ecCodes from ECMWF to read the data. Follow the installation guide provided [here](https://confluence.ecmwf.int/display/ECC/ecCodes+installation).

#### 2.8.1 Installing ecCodes and COSMO definitions

> The GRIB format relies on tables and templates to encode the meta data. Those tables allow for example the mapping of a triplet of

Reviewer: ```suggestion
The GRIB format relies on tables and templates to encode the metadata. Those tables allow for example the mapping of a triplet of
```

numbers encoding a parameter to a descriptive string (`shortName`). The international tables are provided in ecCodes. Additionally ,

> local definitions can be defined by each center. ECMWF specific defintions are also shiped with ecCodes. Some of the provided ICON data

Reviewer: ```suggestion
local definitions can be defined by each center. ECMWF specific definitions are also shiped with ecCodes. Some of the provided ICON data
```

> also requires the local defintions of the COSMO consortium.

Reviewer: ```suggestion
also requires the local definitions of the COSMO consortium.
```
